### PR TITLE
Upgrade to Laravel 5.2

### LINKED ISSUE #1249: Upgrade to Laravel 5.2
Who is ready to do this task?

### PR DESCRIPTION
**Pull request by GrahamCampbell**
Closes #1249.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -20,12 +20,13 @@
...
-        "laravel/framework": "~5.1.26",
+        "laravel/framework": "~5.2.3",
+        "alt-three/bus": "^1.0",
         "alt-three/emoji": "^2.1",
         "alt-three/validator": "^1.4",
-        "barryvdh/laravel-cors": "^0.7",
+        "barryvdh/laravel-cors": "^0.7.3",
         "doctrine/dbal": "^2.5",
-        "fideloper/proxy": "^3.0",
+        "fideloper/proxy": "^3.0.1",
```
Comment (GrahamCampbell) [inline composer.json]: will need to be `^3.1` once the branch alias is sorted out over there

File: app/Foundation/Providers/AppServiceProvider.php [hunk 2]
```diff
@@ -21,7 +21,7 @@ class AppServiceProvider extends ServiceProvider
     /**
      * Boot the service provider.
      *
-     * @param \Illuminate\Bus\Dispatcher $dispatcher
+     * @param \AltThree\Bus\Dispatcher $dispatcher
```
Comment (GrahamCampbell) [inline app/Foundation/Providers/AppServiceProvider.php]: don't get excited yet - i've not even written this yet :trollface:

Comment (jbrooksuk) [inline app/Foundation/Providers/AppServiceProvider.php]: I did wonder ;)

**Comment by GrahamCampbell** [comment]
Currently installed dependencies:

```
alt-three/bus                         dev-master 2465844  An improved command bus for Laravel 5.2
alt-three/emoji                       v2.1.0              An Emoji Parser For Laravel 5
alt-three/testbench                   v1.1.0              Provides Some Testing Traits For Apps
alt-three/validator                   v1.4.0              A Validation Wrapper For Laravel 5
asm89/stack-cors                      dev-master f76d3ff  Cross-origin resource sharing library and stack middleware
bacon/bacon-qr-code                   1.0.0               BaconQrCode is a QR code generator for PHP.
barryvdh/laravel-cors                 v0.7.3              Adds CORS (Cross-Origin Resource Sharing) headers support in your Laravel application
christian-riesen/base32               1.2.2               Base32 encoder/decoder according to RFC 4648
classpreloader/classpreloader         3.0.0               Helps class loading performance by generating a single PHP file containing all of the autoloaded files for a specific use case
dnoegel/php-xdg-base-dir              0.1                 implementation of xdg base directory specification for php
doctrine/annotations                  v1.2.7              Docblock Annotations Parser
doctrine/cache                        v1.5.4              Caching library offering an object-oriented API for many cache backends
doctrine/collections                  v1.3.0              Collections Abstraction library
doctrine/common                       v2.5.2              Common Library for Doctrine projects
doctrine/dbal                         v2.5.2              Database Abstraction Layer
doctrine/inflector                    v1.1.0              Common String Manipulations with regard to casing and singular/plural rules.
doctrine/instantiator                 1.0.5               A small, lightweight utility to instantiate objects in PHP without invoking their constructors
doctrine/lexer                        v1.0.1              Base library for a lexer that can be used in Top-Down, Recursive Descent Parsers.
fideloper/proxy                       dev-patch-5 ad56574 Set trusted proxies for Laravel
filp/whoops                           1.1.10              php error handling for cool kids
fzaninotto/faker                      v1.5.0              Faker is a PHP library that generates fake data for you.
graham-campbell/binput                v3.3.0              Binput Is An Input Protector For Laravel 5
graham-campbell/core                  v4.2.0              Core Provides Some Extra Functionality For Laravel 5
graham-campbell/exceptions            v7.0.0              Provides A Powerful Error Response System For Both Development And Production
graham-campbell/markdown              v5.3.0              Markdown Is A CommonMark Wrapper For Laravel 5
graham-campbell/security              v3.5.0              Security Is A Port Of The Security Class From Codeigniter 3 For Laravel 5
graham-campbell/testbench-core        v1.1.0              TestBench Core Provides Some Testing Functionality For Laravel 5
graham-campbell/throttle              v5.1.0              Throttle Is A Rate Limiter For Laravel 5
guzzlehttp/guzzle                     6.1.1               Guzzle is a PHP HTTP client library
guzzlehttp/promises                   1.0.3               Guzzle promises library
guzzlehttp/psr7                       1.2.1               PSR-7 message implementation
hamcrest/hamcrest-php                 v1.2.2              This is the PHP port of Hamcrest Matchers
jakub-onderka/php-console-color       0.1
jakub-onderka/php-console-highlighter v0.3.2
jenssegers/date                       v3.1.0              A date library to help you work with dates in different languages
jeremeamia/SuperClosure               2.2.0               Serialize Closure objects, including their context and binding
laravel/framework                     v5.2.4              The Laravel Framework.
league/commonmark                     0.12.0              Markdown parser for PHP based on the CommonMark spec
league/flysystem                      1.0.16              Filesystem abstraction: Many filesystems, one API.
mccool/laravel-auto-presenter         4.2.0               A system for auto-decorating models with presenter objects.
mockery/mockery                       0.9.4               Mockery is a simple yet flexible PHP mock object framework for use in unit testing with PHPUnit, PHPSpec or any other testing ...
monolog/monolog                       1.17.2              Sends your logs to files, sockets, inboxes, databases and various web services
mtdowling/cron-expression             v1.0.4              CRON for PHP: Calculate the next or previous run date and determine if a CRON expression is due
nesbot/carbon                         1.21.0              A simple API extension for DateTime.
nikic/php-parser                      v2.0.0              A PHP parser written in PHP
paragonie/random_compat               1.1.4               PHP 5.x polyfill for random_bytes() and random_int() from PHP 7
phpdocumentor/reflection-docblock     2.0.4
phpspec/prophecy                      v1.5.0              Highly opinionated mocking framework for PHP 5.3+
phpunit/php-code-coverage             2.2.4               Library that provides collection, processing, and rendering functionality for PHP code coverage information.
phpunit/php-file-iterator             1.4.1               FilterIterator implementation that filters files based on a list of suffixes.
phpunit/php-text-template             1.2.1               Simple template engine.
phpunit/php-timer                     1.0.7               Utility class for timing
phpunit/php-token-stream              1.4.8               Wrapper around PHP's tokenizer extension.
phpunit/phpunit                       4.8.21              The PHP Unit Testing framework.
phpunit/phpunit-mock-objects          2.3.8               Mock Object library for PHPUnit
pragmarx/google2fa                    v0.7.1              A One Time Password Authentication package, compatible with Google Authenticator.
psr/http-message                      1.0                 Common interface for HTTP messages
psr/log                               1.0.0               Common interface for logging libraries
psy/psysh                             v0.6.1              An interactive shell for modern PHP.
rcrowe/twigbridge                     v0.9.1              Adds the power of Twig to Laravel
roumen/feed                           v2.9.8              A simple feed generator for Laravel.
sebastian/comparator                  1.2.0               Provides the functionality to compare PHP values for equality
sebastian/diff                        1.4.1               Diff implementation
sebastian/environment                 1.3.3               Provides functionality to handle HHVM/PHP environments
sebastian/exporter                    1.2.1               Provides the functionality to export PHP variables for visualization
sebastian/global-state                1.1.1               Snapshotting of global state
sebastian/recursion-context           1.0.2               Provides functionality to recursively process PHP variables
sebastian/version                     1.0.6               Library that helps with managing the version number of Git-hosted PHP projects
simplesoftwareio/simple-qrcode        1.3.2               Simple QrCode is a QR code generator made for Laravel.
swiftmailer/swiftmailer               v5.4.1              Swiftmailer, free feature-rich PHP mailer
symfony/console                       v3.0.0              Symfony Console Component
symfony/css-selector                  v3.0.0              Symfony CssSelector Component
symfony/debug                         v3.0.0              Symfony Debug Component
symfony/dom-crawler                   v3.0.0              Symfony DomCrawler Component
symfony/event-dispatcher              v3.0.0              Symfony EventDispatcher Component
symfony/finder                        v3.0.0              Symfony Finder Component
symfony/http-foundation               v3.0.0              Symfony HttpFoundation Component
symfony/http-kernel                   v3.0.0              Symfony HttpKernel Component
symfony/polyfill-mbstring             v1.0.0              Symfony polyfill for the Mbstring extension
symfony/polyfill-php56                v1.0.0              Symfony polyfill backporting some PHP 5.6+ features to lower PHP versions
symfony/polyfill-util                 v1.0.0              Symfony utilities for portability of PHP codes
symfony/process                       v3.0.0              Symfony Process Component
symfony/routing                       v3.0.0              Symfony Routing Component
symfony/translation                   v3.0.0              Symfony Translation Component
symfony/var-dumper                    v3.0.0              Symfony mechanism for exploring and dumping PHP variables
symfony/yaml                          v3.0.0              Symfony Yaml Component
twig/twig                             v1.23.1             Twig, the flexible, fast, and secure template language for PHP
vlucas/phpdotenv                      v2.1.0              Loads environment variables from `.env` to `getenv()`, `$_ENV` and `$_SERVER` automagically.
```

File: app/Foundation/Providers/AppServiceProvider.php (hunk 2 above)
Comment (GrahamCampbell) [inline app/Foundation/Providers/AppServiceProvider.php]: i've written it now ;)

Comment (jbrooksuk) [inline app/Foundation/Providers/AppServiceProvider.php]: :+1:

**Comment by TheoBearman** [comment]
Is the footer now supposed not to have icons?

**Comment by GrahamCampbell** [comment]
Errr, no? Was that broken by this commit?

**Comment by TheoBearman** [comment]
Maybe not this one but it is currently broken.

**Comment by GrahamCampbell** [comment]
Could you check if it works on https://github.com/cachethq/Cachet/commit/3951f0116a8debbc029ef1c65418e432687c20d5 please?

**Comment by TheoBearman** [comment]
The footer is still broken on 3951f01

**Comment by GrahamCampbell** [comment]
Ok, so it's unrelated to this.

**Comment by GrahamCampbell** [comment]
Could you please create a separate issue. I'm going to remove the comments from this issue.

**Comment by GrahamCampbell** [comment]
Actually, please do not do that. That's NOT a bug. They were intentionally removed in https://github.com/cachethq/Cachet/commit/36f0f9f2cb4b2302a637ec0ed5e4d1131b7f86db.

**Comment by GrahamCampbell** [comment]
They've been removed through all of 2.x.

**Pull request merged**